enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+  per-host storage, including fibre channel settings such as
+  multipath.
+- We want to retain the OpenShift installer's ability to customize a
+  deployment with arbitrary manifests.
+
+### Non-Goals
+
+- A user-driven workflow will be described in a separate enhancement.
+- A user interface for collecting complex inputs such as per-host
+  network settings before running the installation automatically is
+  left to a future enhancement.
+- Changes to the release image mirroring process are out of scope for
+  this enhancement.
+- Eliminating the requirement for a separate image registry to host
+  the mirrored OpenShift image is left to a future enhancement.
+- Describing a mechanism to add and remove nodes from the cluster after
+  it is installed is left to a future enhancement.
+- Generating image formats other than ISO is left to a future
+  enhancement.
+- Booting hosts via PXE is left to a future enhancement.
+- Creating minimal ISO images with a separate rootfs is left to a
+  future enhancement.
+- We are not currently considering replacing the IPI installer or
+  assisted installer SaaS.
+- Describing an end-to-end solution for users without their own
+  orchestration tool is left to a future enhancement.
+- Collecting telemetry data for connected deployments is left to a
+  future enhancement.
+- Details of securing the communication between agents and services to
+  ensure only desired hosts join the cluster are left to a future
+  enhancement.
+- Details of a workflow for deploying in settings with DHCP assignment
+  of IP addresses is left to a future enhancement.
+
+## Proposal
+
+We propose to extend the `openshift-install` binary with a new
+sub-command, `create image` for generating a bootable image containing
+all of the information needed to deploy a cluster.  A command line
+tool avoids issues with prerequisites for deploying a long-running
+service, and can be run on a low-resource host such as a laptop or
+even on a host at a different site before deployment begins.
+
+Most enterprise grade hardware includes a management controller that
+can present an ISO image to a host virtually, allowing the host to
+boot from the image remotely. Hypervisors such as kvm and vSphere also
+support booting from ISOs. Therefore, we will concentrate on that
+format and approach first. Building minimal ISOs with separate root
+filesystems or images that can be used with hosts that PXE boot will
+come later.
+
+We propose to leave up to the user's orchestration tool the
+responsibilities of managing and storing the images, serving them on
+the network, and especially configuring hosts to boot them. This gives
+users the flexibility to perform those operations in any way they
+like, with whatever tool they prefer. A design for an all encompassing
+approach based on tools provided entirely by Red Hat is left for a
+later enhancement.
+
+The `create image` command will need to take inputs similar to what
```

Comment:
@zaneb @celebdor you both had comments/concerns about the details for this aspect of the design. We may want to have a higher bandwidth conversation to work through those.